src/controllers/presentation.ts

```diff
@@ -0,0 +1,243 @@
+	 *         schema:
+	 *           type: boolean
+	 *           default: false
+	 *     requestBody:
+	 *       content:
+	 *         application/x-www-form-urlencoded:
+	 *           schema:
+	 *             $ref: '#/components/schemas/PresentationVerifyRequest'
+	 *         application/json:
+	 *           schema:
+	 *             $ref: '#/components/schemas/PresentationVerifyRequest'
+	 *     responses:
+	 *       200:
+	 *         description: The request was successful.
+	 *         content:
+	 *           application/json:
+	 *             schema:
+	 *               $ref: '#/components/schemas/IVerifyResult'
+	 *       400:
+	 *         $ref: '#/components/schemas/InvalidRequest'
+	 *       401:
+	 *         $ref: '#/components/schemas/UnauthorizedError'
+	 *       500:
+	 *         $ref: '#/components/schemas/InternalError'
+	 */
+	public async verifyPresentation(request: Request, response: Response) {
+		const result = validationResult(request);
+		if (!result.isEmpty()) {
+			return response.status(StatusCodes.BAD_REQUEST).json({ error: result.array()[0].msg });
+		}
+
+		const { presentation, verifierDid, policies, makeFeePayment } = request.body;
+		const verifyStatus = request.query.verifyStatus === 'true';
+		const allowDeactivatedDid = request.query.allowDeactivatedDid === 'true';
+
+		// define identity service strategy setup
+		const identityServiceStrategySetup = new IdentityServiceStrategySetup(response.locals.customer.customerId);
+		// create cheqd presentation from w3c presentation
+		const cheqdPresentation = new CheqdW3CVerifiablePresentation(presentation);
+		// get holder did
+		const holderDid = cheqdPresentation.holder;
+
+		if (!allowDeactivatedDid) {
+			const resolutionResult = await new IdentityServiceStrategySetup().agent.resolveDid(holderDid);
+
+			if (resolutionResult.didDocumentMetadata.deactivated) {
+				return response.status(StatusCodes.BAD_REQUEST).send({
+					error: `${holderDid} is deactivated`,
+				});
+			}
+		}
+
+		if (makeFeePayment) {
+			const feePaymentResult = await cheqdPresentation.makeFeePayment(
+				identityServiceStrategySetup.agent,
+				response.locals.customer
+			);
+			if (feePaymentResult.error) {
+				return response.status(StatusCodes.BAD_REQUEST).json({
+					checked: false,
+					error: `verify: payment: error: ${feePaymentResult.error}`,
+				});
+			}
+		}
+
+		try {
+			const result = await identityServiceStrategySetup.agent.verifyPresentation(
+				cheqdPresentation,
+				{
+					verifyStatus,
+					policies,
+					domain: verifierDid,
+				},
+				response.locals.customer
+			);
+			if (result.error) {
+				return response.status(StatusCodes.BAD_REQUEST).json({
+					verified: result.verified,
+					error: result.error.message,
+				});
+			}
+			return response.status(StatusCodes.OK).json(result);
+		} catch (error) {
+			return response.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+				error: `Internal error: ${(error as Error)?.message || error}`,
+			});
```

Comment:
This is good as a catch-all error handling, but it might be misleading for client-originating issues, notably 400s range, unauthorised being the most common in decryption flow.

Suggesting the same way errors are handled in revocation controller, but free to adapt to your coding style, if necessary:

```suggestion
			// define error
			const errorRef = error as Record<string, unknown>;

			// handle doesn't meet condition
			if (errorRef?.errorCode === 'NodeAccessControlConditionsReturnedNotAuthorized')
				return response.status(StatusCodes.UNAUTHORIZED).json({
					checked: false,
					error: `check: error: ${
						errorRef?.message
							? 'unauthorised: decryption conditions are not met'
							: (error as Record<string, unknown>).toString()
					}`,
				} satisfies CheckStatusListUnsuccessfulResponseBody);

			// handle incorrect access control conditions
			if (errorRef?.errorCode === 'incorrect_access_control_conditions')
				return response.status(StatusCodes.BAD_REQUEST).json({
					checked: false,
					error: `check: error: ${
						errorRef?.message
							? 'incorrect access control conditions'
							: (error as Record<string, unknown>).toString()
					}`,
				} satisfies CheckStatusListUnsuccessfulResponseBody);

			// return catch-all error
			return response.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
				checked: false,
				error: `Internal error: ${errorRef?.message || errorRef.toString()}`,
			} satisfies CheckStatusListUnsuccessfulResponseBody);
```